src/database/hooks/useSaveAnonChatHook.ts

```diff
@@ -19,17 +19,49 @@ const useSaveAnonChatHook = () => {
   const {goToChatScreen} = useChatUtilsHook();
   const {signedProfileId, anonProfileId} = useUserAuthHook();
 
-  const helperFindChatById = async (object: InitAnonymousChatData): Promise<ChannelListSchema> => {
+  const helperFindChatById = async (
+    object: InitAnonymousChatData
+  ): Promise<ChannelListSchema | null> => {
     const channelId = object?.message?.cid?.replace('messaging:', '');
     const channelList = await ChannelListSchema.getById(localDb, channelId);
 
     const channelListSchema = ChannelListSchema.fromDatabaseObject(channelList);
     return channelListSchema;
   };
 
+  const saveUserAndChannelListMember = async (channelId, member) => {
+    try {
+      const userMember = UserSchema.fromInitAnonymousChatAPI(member, channelId);
+      await userMember.saveOrUpdateIfExists(localDb);
+    } catch (e) {
+      console.log('error saveUserAndChannelListMember', e);
+    }
+  };
+
   const helperGoToAnonymousChat = async (object: InitAnonymousChatData, from: string) => {
-    const channelListSchema = await helperFindChatById(object);
-    goToChatScreen(channelListSchema, from);
+    try {
+      const channelListSchema = await helperFindChatById(object);
+      goToChatScreen(channelListSchema as ChannelList, from);
+    } catch (e) {
+      console.log('error helperGoToAnonymousChat', e);
+    }
+  };
+
+  const helperSaveChannel = async (object: ModifyAnonymousChatData, type: string) => {
+    const channelId = object?.message?.cid?.replace('messaging:', '');
+    let channelList = await ChannelListSchema.getById(localDb, channelId);
+
+    if (!channelList) {
+      channelList = ChannelListSchema.fromInitAnonymousChatAPI(
+        object,
+        type === SIGNED ? 'PM' : 'ANON_PM'
+      );
+    }
+
+    channelList.description = object?.message?.text;
+    channelList.lastUpdatedAt = new Date().toISOString();
+    channelList.lastUpdatedBy = object?.message?.user?.id;
+    await channelList.saveIfLatest(localDb);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The error handling in the `saveUserAndChannelListMember` and `helperGoToAnonymousChat` functions could be improved. Currently, if an error occurs, it's logged to the console but not propagated further. This could lead to silent failures that are hard to debug. Consider throwing the error after logging it or returning a rejected Promise.

```diff
  const saveUserAndChannelListMember = async (channelId, member) => {
    try {
      const userMember = UserSchema.fromInitAnonymousChatAPI(member, channelId);
      await userMember.saveOrUpdateIfExists(localDb);
    } catch (e) {
      console.log('error saveUserAndChannelListMember', e);
+      throw e;
    }
  };

   const helperGoToAnonymousChat = async (object: InitAnonymousChatData, from: string) => {
    try {
      const channelListSchema = await helperFindChatById(object);
      goToChatScreen(channelListSchema as ChannelList, from);
    } catch (e) {
      console.log('error helperGoToAnonymousChat', e);
+      throw e;
    }
  };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->